### PR TITLE
Move opts to internal/opts and fork docker filters

### PR DESCRIPTION
**Pull request by vrothberg**
The dependency on docker's filter package prevents us from using newer
versions of docker/moby where the package's API has changed.  The filter
package is small enough to justify maintaining a small fork of it. Also,
since the opts package is only used for the `containers-storage` binary
and is not part of the actual storage library, move it to the ./internal
directory to make it clearer that this is an internal API.

Fixes: #450
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@rhatdan @nalind @TomSweeneyRedHat @giuseppe PTAL

File: internal/opts/parse.go [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 	"encoding/json"
 	"errors"
 	"regexp"
 	"strings"
-
-	"github.com/docker/docker/api/types/versions"
 )
 
-// Args stores a mapping of keys to a set of multiple values.
+// args stores a mapping of keys to a set of multiple values.
```
Comment (rhatdan) [inline internal/opts/parse.go]: Nit: Why change case?

**Comment by rhatdan** [comment]
LGTM except for one Nit.

File: internal/opts/parse.go (hunk 1 above)
Comment (vrothberg) [reply internal/opts/parse.go]: Good catch. I was playing with the idea of making those types private but had to revert it at some point. I missed this comment.

Comment (vrothberg) [reply internal/opts/parse.go]: Fixed now.

**Comment by TomSweeneyRedHat** [comment]
LGTM but would like a more storage savy head nod.

**Comment by giuseppe** [comment]
LGTM.  Do we need to bump the major version?

**Comment by vrothberg** [comment]
> LGTM. Do we need to bump the major version?

It depends if we consider this to be a bug fix or not. I think it's a bug that we exposed an API that it's used by a binary for testing and I personally don't consider this package to be part of the storage library. From the user perspective, I don't know of any tool that depends on this package.

**Comment by rhatdan** [comment]
I would say this is a bug and should not need a bump.

**Pull request merged**